native-land-digital/native-land-web-client
# Adds stylelint, precommit linting steps

## Summary
This includes work described in #8.

## Testing Notes
1. Pull down this branch and run `yarn` to install all dependencies
1. Make changes that you would expect to be auto-corrected in a linter in any `.tsx` file.
  1. Example: change double quotes to single quotes and remove some semi-colons
1. Run `git add -A`, `yarn run precommit`. Verify that the file was run on the linter and was auto-corrected
1. Make changes that you would expect to generate an error in a linter in any `.tsx` file.
  1. Example: Export an object literal from a `.tsx` file
1. Attempt to commit the code with `git add -A` and `git commit`
1. Verify the commit is aborted due to the linting steps and errors from lint-staged
1. Create a `.scss` file and write some generic CSS attributes.
1. Run `git add -A`, `yarn run precommit`. Verify that the file was run on the linter and was auto-corrected for proper CSS attribute order

## Caveats / Concerns
1. Stylelint was configured to not allow CSS attributes that arent supported by the browser config. Currently, the browser config is set to `latest Chrome version`. This means basically any CSS is supported. We may want to adjust this setting if we want to support more browsers officially
